aws/resource_aws_backup_vault.go

```diff
@@ -0,0 +1,120 @@
+package aws
+
+import (
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/validation"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/backup"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsBackupVault() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsBackupVaultCreate,
+		Read:   resourceAwsBackupVaultRead,
+		Update: resourceAwsBackupVaultUpdate,
+		Delete: resourceAwsBackupVaultDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`[a-z0-9\-]+`), "must contain alphanumeric characters or underscores"),
+			},
+			"tags": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"kms_key_arn": {
```

Comment:
For proper drift detection, this attribute should have `d.Set("kms_key_arn", ...)` called during `resourceAwsBackupVaultRead`, e.g.

```go
d.Set("kms_key_arn", resp.EncryptionKeyArn)
```

Although since the API names this `EncryptionKeyArn`, is there any reason to not following the API naming by calling this attribute `encryption_key_arn`? Maybe they plan on supporting non-KMS keys.

Reply:
This is one of those weird resources where AWS will auto-populate this attribute with the default account KMS key if you don't provide a specific key.  Based on that piece of information, I'm comfortable with keeping the name how it is since that implies to me that they intend to focus on KMS.